pkg/services/object/server.go

```diff
@@ -182,13 +214,27 @@ func (s *server) Delete(ctx context.Context, req *protoobject.DeleteRequest) (*p
 	}
 
 	t := time.Now()
+	if err := signature.VerifyServiceMessage(delReq); err != nil {
+		return s.makeStatusDeleteResponse(t, err)
+	}
 	resp, err := s.srv.Delete(ctx, delReq)
-	s.pushOpExecResult(stat.MethodObjectDelete, err, t)
 	if err != nil {
-		return nil, err
+		return s.makeStatusDeleteResponse(t, err)
 	}
 
-	return resp.ToGRPCMessage().(*protoobject.DeleteResponse), nil
+	return s.signDeleteResponse(t, util.StatusOKErr, resp.ToGRPCMessage().(*protoobject.DeleteResponse))
+}
+
+func (s *server) signHeadResponse(err error, startTime time.Time, resp *protoobject.HeadResponse) (*protoobject.HeadResponse, error) {
+	resp = util.SignResponse(&s.signer, resp, v2object.HeadResponse{})
+	s.pushOpExecResult(stat.MethodObjectHead, err, startTime)
+	return resp, nil
+}
+
+func (s *server) makeStatusHeadResponse(startTime time.Time, err error) (*protoobject.HeadResponse, error) {
```

Comment:
Can we have a generic version of it with two types?

Reply:
didnt get what do u mean by "two types". What signature would u expect?

Reply:
Likely not possible without an additional `InitFromMetaHeader` for each `protoobject.*`. The idea is to have `*protoobject.Smth` as a generic type, but it also requires `v2object.Smth` companion. Which is handled in `SignResponse` already with R, RV2 and RV2PTR, so our hypothetical `makeGenericStatusResponse` would require all of them plus one more type for `*protoobject.Smth`. Can be somewhat messy.

Reply:
generic struct fields would also be helpful, but Go does not support them. If it becomes, and when v2 leaves, we can simplify this code